apps/web/pages/event-types/[type]/index.tsx

```diff
@@ -321,6 +322,47 @@ const EventTypePage = (props: EventTypeSetupProps) => {
                   teamName: z.string().optional(),
                 })
                 .passthrough()
+                .superRefine((val, ctx) => {
+                  if (val?.link) {
+                    const link = val.link;
+                    const eventLocationType = getEventLocationType(val.type);
+                    if (
+                      eventLocationType &&
+                      !eventLocationType.default &&
+                      eventLocationType.linkType === "static" &&
+                      eventLocationType.urlRegExp
+                    ) {
+                      const valid = z
+                        .string()
+                        .regex(new RegExp(eventLocationType.urlRegExp))
+                        .safeParse(link).success;
+
+                      if (!valid) {
+                        const sampleUrl = eventLocationType.organizerInputPlaceholder;
+                        ctx.addIssue({
+                          code: z.ZodIssueCode.custom,
+                          path: [eventLocationType?.defaultValueVariable ?? "link"],
+                          message: t("invalid_url_error_message", {
+                            label: eventLocationType.label,
+                            sampleUrl: sampleUrl ?? "https://cal.com",
+                          }),
+                        });
+                      }
+                      return;
+                    }
+
+                    const valid = z.string().url().optional().safeParse(link).success;
+
+                    if (!valid) {
+                      ctx.addIssue({
+                        code: z.ZodIssueCode.custom,
+                        path: [eventLocationType?.defaultValueVariable ?? "link"],
+                        message: `Invalid URL`,
+                      });
+                    }
+                  }
+                  return;
+                })
```

Comment:
can you explain what all of this is? looks pretty funky

Reply:
@PeerRich this code is from editLocationDialog.tsx. we are just validating the url here . We already have regex of some location options that vaildates if url is valid and if it not then displaying error message with the sampleUrl that we are expecting

Reply:
do we wanna extract this into a helper function if we use it in two places?

Reply:
or is this the code we removed but are re-adding?

Reply:
actually There are some differences like this line isn't required in EditLocationDialog

` const eventLocationType = getEventLocationType(val.type);
` 
 and path ( path: [eventLocationType?.defaultValueVariable ?? "link"]) is also not required in EditLocationDialog.

Reply:
so we are ok merging this?

Reply:
yeah